Build of libbigtable_client_testing fails on macOS Big Sur + CMake / Ninja

**What component of `google-cloud-cpp` is this related to?**
Bigtable

**Describe the bug**
I'm trying to build the project from the [nixpkgs package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/google-cloud-cpp/default.nix), but it errors on macOS with:
```
FAILED: google/cloud/bigtable/libbigtable_client_testing.dylib
: && /nix/store/18i795dcmzx2jlc7560nf65qybp0y41p-clang-wrapper-11.1.0/bin/clang++ -O3 -DNDEBUG  -dynamiclib -Wl,-headerpad_max_install_names  -o google/cloud/bigtable/libbigtable_client_testing.dylib -install_nam
Undefined symbols for architecture x86_64:
  "google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher::DescribeTo(std::__1::basic_ostream<char, std::__1::char_traits<char> >*) const", referenced from:
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::StatusOr<google::bigtable::admin::v2::Table> const&>::DescribeTo(std::__1
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::Status const&>::DescribeTo(std::__1::basic_ostream<char, std::__1::char_t
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::StatusOr<google::cloud::bigtable::v1_38_0::Row> const&>::DescribeTo(std::
  "google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher::MatchAndExplain(google::cloud::v1_38_0::Status const&, testing::MatchResultListener*) const", referenced from:
      bool google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher::MatchAndExplain<google::bigtable::admin::v2::Table>(google::cloud::v1_38_0::StatusOr<google::bigtable::admin::v2::Table> const&, testing:
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::Status const&>::MatchAndExplain(google::cloud::v1_38_0::Status const&, te
      bool google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher::MatchAndExplain<google::cloud::bigtable::v1_38_0::Row>(google::cloud::v1_38_0::StatusOr<google::cloud::bigtable::v1_38_0::Row> const&, te
  "google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher::DescribeNegationTo(std::__1::basic_ostream<char, std::__1::char_traits<char> >*) const", referenced from:
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::StatusOr<google::bigtable::admin::v2::Table> const&>::DescribeNegationTo(
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::Status const&>::DescribeNegationTo(std::__1::basic_ostream<char, std::__1
      testing::PolymorphicMatcher<google::cloud::v1_38_0::testing_util_internal::StatusIsMatcher>::MonomorphicImpl<google::cloud::v1_38_0::StatusOr<google::cloud::bigtable::v1_38_0::Row> const&>::DescribeNegation
  "typeinfo for google::cloud::v1_38_0::testing_util::IntegrationTest", referenced from:
      typeinfo for google::cloud::bigtable::testing::TableIntegrationTest in table_integration_test.cc.o
  "typeinfo for google::cloud::v1_38_0::testing_util::FakeCompletionQueueImpl", referenced from:
      google::cloud::bigtable::testing::TableTestFixture::TableTestFixture(google::cloud::v1_38_0::CompletionQueue const&) in table_test_fixture.cc.o
ld: symbol(s) not found for architecture x86_64
```

I tried to debug a bit and added the `google_cloud_cpp_testing` target to the link libraries of `bigtable_client_testing`. This improves the situation somewhat, but one missing symbol remains:

```
FAILED: google/cloud/bigtable/libbigtable_client_testing.dylib
: && /nix/store/18i795dcmzx2jlc7560nf65qybp0y41p-clang-wrapper-11.1.0/bin/clang++ -O3 -DNDEBUG -mmacosx-version-min=10.12 -dynamiclib -Wl,-headerpad_max_install_names  -o google/cloud/bigtable/libbigtable_client_testing.dylib -install_name libbigtable_client_testing.dylib google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/cleanup_stale_resources.cc.o google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/embedded_server_test_fixture.cc.o google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/inprocess_data_client.cc.o google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/random_names.cc.o google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/table_integration_test.cc.o google/cloud/bigtable/CMakeFiles/bigtable_client_testing.dir/testing/table_test_fixture.cc.o  google/cloud/testing_util/libgoogle_cloud_cpp_testing.dylib  /nix/store/xxr7qcj4rzdk11lm5jbd76rh9295myvj-gtest-1.11.0/lib/libgmock_main.1.11.0.dylib  /nix/store/xxr7qcj4rzdk11lm5jbd76rh9295myvj-gtest-1.11.0/lib/libgmock.1.11.0.dylib  /nix/store/xxr7qcj4rzdk11lm5jbd76rh9295myvj-gtest-1.11.0/lib/libgtest.1.11.0.dylib  /nix/store/vqkhd7yv9v7x408b4viw9dlv3a23dfvy-protobuf-3.19.3/lib/libprotobuf.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_failure_signal_handler.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_examine_stack.2103.0.1.dylib  google/cloud/bigtable/libgoogle_cloud_cpp_bigtable.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_bigtable_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_routing_protos.1.38.0.dylib  google/cloud/libgoogle_cloud_cpp_grpc_utils.1.38.0.dylib  google/cloud/libgoogle_cloud_cpp_common.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_longrunning_operations_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_rpc_status_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_iam_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_iam_v1_iam_policy_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_client_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_field_behavior_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_resource_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_iam_v1_options_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_iam_v1_policy_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_annotations_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_api_http_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_type_expr_protos.1.38.0.dylib  external/googleapis/libgoogle_cloud_cpp_rpc_error_details_protos.1.38.0.dylib  /nix/store/ajvckv8m8vgixl8p24p298h28x2gczc0-grpc-1.43.0/lib/libgrpc++.1.43.0.dylib  /nix/store/ajvckv8m8vgixl8p24p298h28x2gczc0-grpc-1.43.0/lib/libgrpc.21.0.0.dylib  /nix/store/2r6pwl5f1s8xhq9d3l4g35cxl7fw5dbg-zlib-1.2.11/lib/libz.dylib  /nix/store/z9ns6nbq04rw7q6c1n34m319q0rnan1j-c-ares-1.18.1/lib/libcares.dylib  /nix/store/ajvckv8m8vgixl8p24p298h28x2gczc0-grpc-1.43.0/lib/libaddress_sorting.21.0.0.dylib  -L/nix/store/0jv8ww7sss89akb2gf9cxxqqbbbncdsa-re2-2021-11-01/lib  -pthread  -lre2  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_raw_hash_set.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_hashtablez_sampler.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_exponential_biased.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_hash.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_city.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_wyhash.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_statusor.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_bad_variant_access.2103.0.1.dylib  /nix/store/ajvckv8m8vgixl8p24p298h28x2gczc0-grpc-1.43.0/lib/libgpr.21.0.0.dylib  /nix/store/ajvckv8m8vgixl8p24p298h28x2gczc0-grpc-1.43.0/lib/libupb.21.0.0.dylib  -lm  -lpthread  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_status.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_cord.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_bad_optional_access.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_str_format_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_synchronization.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_symbolize.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_demangle_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_stacktrace.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_debugging_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_graphcycles_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_malloc_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_time.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_strings.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_throw_delegate.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_strings_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_base.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_spinlock_wait.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_civil_time.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_int128.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_time_zone.2103.0.1.dylib  -framework  CoreFoundation  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_raw_logging_internal.2103.0.1.dylib  /nix/store/30yf2gvcvvjl25n02ja509nl6vsqfi1d-abseil-cpp-20210324.2/lib/libabsl_log_severity.2103.0.1.dylib  /nix/store/h65q0gz96ghs734cziigida2ppyib9ba-openssl-1.1.1m/lib/libssl.dylib  /nix/store/h65q0gz96ghs734cziigida2ppyib9ba-openssl-1.1.1m/lib/libcrypto.dylib  -framework CoreFoundation  /nix/store/vqkhd7yv9v7x408b4viw9dlv3a23dfvy-protobuf-3.19.3/lib/libprotobuf.dylib && :
Undefined symbols for architecture x86_64:
  "typeinfo for google::cloud::v1_38_0::testing_util::FakeCompletionQueueImpl", referenced from:
      google::cloud::bigtable::testing::TableTestFixture::TableTestFixture(google::cloud::v1_38_0::CompletionQueue const&) in table_test_fixture.cc.o
ld: symbol(s) not found for architecture x86_64
```

**Operating system:**
macOS Big Sur

**What compiler and version are you using?**
```
clang version 11.1.0
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /nix/store/a65laibvivl4ahwm16h93yyi28z73lnv-clang-11.1.0/bin
```

**What version of `google-cloud-cpp` are you using?**
Tag v1.38.0.

**Additional context**
Add any other context about the problem here.

--- a/google/cloud/storage/CMakeLists.txt
+++ b/google/cloud/storage/CMakeLists.txt
@@ -450,6 +450,7 @@ if (BUILD_TESTING)
         PUBLIC absl::memory
                google-cloud-cpp::experimental-storage-grpc
                google-cloud-cpp::storage
+               google_cloud_cpp_testing
                nlohmann_json::nlohmann_json
                GTest::gmock_main
                GTest::gmock

